src/main/java/com/bookstore/service/impl/BookServiceImpl.java

```diff
@@ -12,14 +16,26 @@
 public class BookServiceImpl implements BookService {
 
     private final BookRepository bookRepository;
+    private final BookMapper bookMapper;
 
     @Override
-    public Book save(Book book) {
-        return bookRepository.save(book);
+    public BookResponseDto save(BookRequestDto bookRequestDto) {
+        Book book = bookMapper.mapToEntity(bookRequestDto);
+        bookRepository.save(book);
+        return bookMapper.mapToDto(book);
     }
 
     @Override
-    public List<Book> findAll() {
-        return bookRepository.findAll();
+    public List<BookResponseDto> findAll() {
+        return bookRepository.findAll().stream()
+                .map(bookMapper::mapToDto)
+                .toList();
+    }
+
+    @Override
+    public BookResponseDto findById(Long id) {
+        Book book = bookRepository.findById(id).orElseThrow(
+                () -> new EntityNotFoundException(String.format("Book with id: %s not found", id)));
```

Comment:
use Optional#map to inline code
```suggestion
        Book book = bookRepository.findById(id).orElseThrow(
                () -> new EntityNotFoundException(String.format("Book with id: %s not found", id)));
```